### ISSUE TITLE
Node crushes with the message " No space left on device : 28 : Can't create temporary file"

### ISSUE THREAD

**Issue by begetan** [ton-blockchain/ton#15]
After 3 days of the run, my node has stopped to work. Here are the log messages:

```bash
[ 2][t 1][1568254354.870851755][check-proof.cpp:46][!checkproof][&VALIDATOR_WARNING]    aborting check proof for [ w=-1 s=9223372036854775808 seq=181459 WuEW/fhVUp1f2xS2J5iZVzkK5J1q0aaL6No48/rjLEU= 6xdgUDLg691ayMEW++Ld0WK/nXmTUQhNm3ydB1E1FVs= ] query: [PosixError : No space left on device : 28 : Can't create temporary file "/var/ton-work/db/files/tmp/tmpXXXX7pNrZj"]
[ 3][t 1][1568254354.871072054][download-block.cpp:78][!downloadnext][&FULL_NODE_NOTICE]        failed to download block [ w=-1 s=9223372036854775808 seq=181459 WuEW/fhVUp1f2xS2J5iZVzkK5J1q0aaL6No48/rjLEU= 6xdgUDLg691ayMEW++Ld0WK/nXmTUQhNm3ydB1E1FVs= ] from T+dQck1Slc+Nyz7LxSpCxgIDGYilPtp5Is8gi7k1oeY=: [PosixError : No space left on device : 28 : Can't create temporary file "/var/ton-work/db/files/tmp/tmpXXXX7pNrZj"]
[ 2][t 1][1568254354.871119976][full-node-shard.cpp:121][!downloadnext][&FULL_NODE_WARNING]     failed to download next block after [ w=-1 s=9223372036854775808 seq=181458 /fi5NXrPIVCd4xjnauas2ekUGq2uVl9D/RY5CL9swG4= qdJQvKjDtJGLQEv2wcyWU3NIm8JPgsbgdg3hO2we9R8= ]: [PosixError : No space left on device : 28 : Can't create temporary file "/var/ton-work/db/files/tmp/tmpXXXX7pNrZj"]
[ 0][t 1][1568254354.871142864][block-handle.hpp:483][!downloadnext][&!need_flush()]    flags=188579
````

Here is the disk state:
```bash
/var/ton-work/log# df -h
Filesystem      Size  Used Avail Use% Mounted on
udev            1.9G     0  1.9G   0% /dev
tmpfs           385M  672K  385M   1% /run
/dev/sda1        38G   26G   11G  72% /
tmpfs           1.9G     0  1.9G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           1.9G     0  1.9G   0% /sys/fs/cgroup
tmpfs           385M     0  385M   0% /run/user/0
```

May it be an OS limit for the number of the files, open files and so on?

At the moment I see about 131.000 subdirectories in db directories. Most of the directories are empty. Indexing of such amount of the file structures is not efficient I guess.

The system is run out of inodes because of a lot amount of the directories 
```bash
 df -i
Filesystem      Inodes   IUsed  IFree IUse% Mounted on
udev            488900     405 488495    1% /dev
tmpfs           492643     583 492060    1% /run
/dev/sda1      2448000 2448000      0  100% /
tmpfs           492643       1 492642    1% /dev/shm
tmpfs           492643       3 492640    1% /run/lock
tmpfs           492643      18 492625    1% /sys/fs/cgroup
tmpfs           492643       1 492642    1% /run/user/0

/var/ton-work/db# ls -1 -R | wc
2649022 2517418 149932828
```
There are more tnan 2.6 millions of files and directories at the moment, so system run out of the inodes. Unfortunately, it is not possible to change the inodes configuration after creating ext4 filesystem.

**Comment by begetan** [comment]
I've created a separate partition with bigger size and format with the XFS filesystem which allows adjusting a number of inodes

**Comment by begetan** [comment]
This issue probably was fixed after Testnet reset at 19.09.2019 with the current build.

```
# df -h | grep /var/ton-work
/dev/sdb1       100G   19G   82G  19% /var/ton-work
# df -i | grep /var/ton-work
/dev/sdb1      52428288     175 52428113    1% /var/ton-work
```
There are only several  subdirectories inside `db/files/files/` opposing milions before.
